test/unit/org/apache/cassandra/auth/CassandraRoleManagerTest.java

```diff
@@ -85,4 +103,75 @@ private void fetchRolesAndCheckReadCount(IRoleManager roleManager, RoleResource
         long after = getRolesReadCount();
         assertEquals(granted.size(), after - before);
     }
+
+    @Test
+    public void confirmFastRoleSetup() throws Exception
+    {
+        IRoleManager roleManager = new AuthTestUtils.LocalCassandraRoleManager();
+        roleManager.setup();
+        for (RoleResource r : ALL_ROLES)
+            roleManager.createRole(AuthenticatedUser.ANONYMOUS_USER, r, new RoleOptions());
+
+        CassandraRoleManager crm = new CassandraRoleManager();
+
+        assertEquals(CassandraRoleManager.hasExistingRoles(), true);
+        assertEquals(crm.isClusterReady(), false);
```

Comment:
nit: could use `assertTrue()` and `assertFalse()` respectively (although perhaps that's less important than throwing an explicit failure message in there)

Reply:
Fleshed out a bit and swapped to the less tautological options.